test/testcmdlineparser.cpp

```diff
@@ -368,14 +369,26 @@ class TestCmdlineParser : public TestFixture {
         ASSERT_EQUALS("", GET_REDIRECT_OUTPUT);
     }
 
-    void showversion() {
+    void version() {
         REDIRECT;
         const char * const argv[] = {"cppcheck", "--version"};
         ASSERT(parser->parseFromArgs(2, argv));
-        ASSERT_EQUALS(true, parser->getShowVersion());
-        ASSERT_EQUALS("", logger->str()); // version is not actually shown
+        ASSERT_EQUALS("Cppcheck 2.13 dev\n", logger->str());
     }
 
+    void versionWithCfg() {
+        REDIRECT;
+        ScopedFile file("cppcheck.cfg",
+                "{\n"
+                "\"productName\": \"The Product\""
+                "}\n");
+        const char * const argv[] = {"cppcheck", "--version"};
+        ASSERT(parser->parseFromArgs(2, argv));
+        ASSERT_EQUALS("The Product\n", logger->str()); // TODO: include version?
+    }
+
+    // TODO: test extraVersion
```

Comment:
That appears to be unused. Does premium make use of that?

Reply:
No premium does not use extraVersion. I don't remember why we added it maybe it's for "dev" or sha or something..